ember-file-upload/src/system/http-request.ts

```diff
@@ -82,15 +82,20 @@ export default class HTTPRequest {
       aborted.resolve();
     });
 
-    this.request.onloadstart =
-      this.request.onprogress =
-      this.request.onloadend =
-        bind(this, function (evt) {
-          this.onprogress?.(evt);
-        });
+    this.request.onloadstart = bind(this, function (evt) {
+      this.onprogress?.(evt);
+    });
+    this.request.onprogress = bind(this, function (evt) {
+      this.onprogress?.(evt);
+    });
+    this.request.onloadend = bind(this, function (evt) {
+      this.onprogress?.(evt);
+    });
 
     if (this.request.upload) {
+      this.request.upload.onloadstart = this.request.onprogress;
       this.request.upload.onprogress = this.request.onprogress;
+      this.request.upload.onloadend = this.request.onprogress;
     }
```

Comment:
Note these are still intentionally wired up to call `onprogress` only. In the actual bugfix they will each call their own method.